common/processtracker_windows.cpp

```diff
@@ -0,0 +1,429 @@
+/*
+  processtracker_windows.cpp
+
+  This file is part of GammaRay, the Qt application inspection and
+  manipulation tool.
+
+  Copyright (C) 2016-2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Filipe Azevedo <filipe.azevedo@kdab.com>
+
+  Licensees holding valid commercial KDAB GammaRay licenses may use this file in
+  accordance with GammaRay Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// A said crossplatform way (using ptrace) is available at
+// https://forum.juce.com/t/detecting-if-a-process-is-being-run-under-a-debugger/2098
+
+#include "processtracker_windows.h"
+
+#include <QCoreApplication>
+#include <QLibrary>
+#include <QDebug>
+
+#include <qt_windows.h>
+// Processthreadsapi.h on Windows 8 and Windows Server 2012
+//#include <tlhelp32.h>
+
+#include <type_traits>
+#include <memory>
+
+namespace {
+// C++ wrappers / helpers
+
+struct LocalHandleDeleter
+{
+    void operator()(HANDLE handle)
+    {
+        if (handle) {
+            CloseHandle(handle);
+            handle = INVALID_HANDLE_VALUE;
+        }
+    }
+};
+
+// Non pointer HANDLE helper
+using LHANDLE = std::remove_pointer<HANDLE>::type;
+// HANDLE holder helper
+using LocalHandlePtr = std::unique_ptr<LHANDLE, LocalHandleDeleter>;
+
+class LocalBuffer
+{
+private:
+    BYTE* m_data = nullptr;
+    DWORD m_size = 0;
+
+    void allocate(DWORD size) {
+        m_data = (BYTE*)LocalAlloc(LMEM_FIXED, size);
+        if (m_data != nullptr) {
+            m_size = size;
+        }
+    }
+
+    void free() {
+        if (m_data != nullptr) {
+            LocalFree(m_data);
+            m_data = nullptr;
+            m_size = 0;
+        }
+    }
+
+public:
+    LocalBuffer(DWORD size) {
+        allocate(size);
+    }
+
+    ~LocalBuffer() {
+        free();
+    }
+
+    LocalBuffer(const LocalBuffer &other) = delete;
+    LocalBuffer &operator=(const LocalBuffer &other) = delete;
+
+    bool isValid() const {
+        return m_data != nullptr;
+    }
+
+    DWORD size() const {
+        return m_size;
```

Comment:
If allocate() failed m_size will be different from 0 even if m_data is null, so maybe either set m_size to 0 there or check m_data here.